source/adapters/level_zero/kernel.cpp

```diff
@@ -229,6 +233,12 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
   if (IndirectAccessTrackingEnabled)
     Queue->KernelsToBeSubmitted.push_back(Kernel);
 
+  if (!Queue->AppendedEvents) {
+    Queue->AppendedEvents = new ze_event_handle_t[Queue->totalCmdListCount - 1];
```

Comment:
So, I would make this obvious or leave a comment here saying that the "AppendedEvents" are device scope.

ie maybe change to DeviceAppendedEvents or something like that.